styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java

```diff
@@ -169,6 +193,20 @@ WorkflowInstanceExecutionData executionData(WorkflowInstance workflowInstance) t
     return WorkflowInstanceExecutionData.fromEvents(events);
   }
 
+  private List<WorkflowInstanceExecutionData> executionData(
+      Set<WorkflowInstance> workflowInstancesSet) {
+    return workflowInstancesSet.parallelStream()
+        .map(workflowInstance -> {
+          try {
+            return executionData(workflowInstance);
+          } catch (IOException e) {
+            throw Throwables.propagate(e);
+          }
+        })
+        .sorted(WorkflowInstanceExecutionData.COMPARATOR)
```

Comment:
This is for another PR, but it's confusing that we sort the instances on timestamp when the order appears guaranteed by BigTable. I was thinking this could relate to `parallelStream` but the collection order in Java 8 streams are guaranteed to be the same as the original, if the streams are created from Lists (see [these docs](https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html#Ordering)). Maybe it's a parallel optimization thing, but in that case the speed benefit is probably not worth the mental burden.